### ISSUE TITLE
Automatically add override flags

### ISSUE THREAD

**Issue by zdavatz** [zdavatz/oddb.org#237]
1. Is it possible to automatically add the override flags to `etc/barcode_to_text_info.yml`
2. <img width="538" alt="PNG-Bild" src="https://github.com/zdavatz/oddb.org/assets/4953/eaf4109d-0de5-4195-b2e2-0dc0dd6937d4">

**Comment by zdavatz** [comment]
doing this manually.

**Comment by zdavatz** [comment]
the goal would be to add the missing overrides and restart the job.

**Comment by zdavatz** [comment]
this needs to be automated: https://github.com/zdavatz/oddb.org/commit/3770c914076193f7e24f00f4d47d578465f356c9

**Comment by ngiger** [comment]
The problem is that the emitted file is not a valid YAML file, as a line like `7680373970327_pi_de: 'Ventolin® 1,25 mg/2,5 mL gebrauchsfertige Lösung für Aerosolgeräte'` cannot be parsed.
A <'> must be escaped. Or we must emit the name with enclosing <">
This leads to a problem when reading the yaml in import_daily

**Comment by zdavatz** [comment]
the problem comes from one of the updated gem files:

https://github.com/zdavatz/oddb.org/commit/8c81580776539e27f26dbb8f0eb2f9bdddaaf985

**Comment by ngiger** [comment]
The line breaks in the generated file come from the default line width of the psych emitter.
Can be fixed with the following ruby file `fix_barcode.rb`
```
#!/usr/bin/env ruby
require 'yaml'
outname = 'etc/barcode_to_text_info.yml'
content = File.read(outname)
overrides = YAML.load content.gsub(/(i_(it|de|fr): )'|'$/, "\\1")
overrides.each{ |id, name|
                name.gsub!("\n", '')
                }
File.open(outname, "w") { |file| file.write(Psych.dump(overrides, {line_width: 120})) }

```

**Comment by zdavatz** [comment]
This can go into the 3.3 branch, but only after oddb.org runs smoothly on Ruby 3.3

**Comment by ngiger** [comment]
Fixed this problem (in ruby 3.2), as it was easy. Added a good test case for it, too.

**Comment by zdavatz** [comment]
```
7680450440255_pi_de: 'Dr. Reckeweg R14 Quieta, Tropfen zum Einnehmen # != override '
7680450440255_pi_fr: 'Dr. Reckeweg R14 Quieta, Gouttes buvables # != override '
7680450520360_pi_de: 'Dr. Reckeweg® R23 Nosoderm, Tropfen zum Einnehmen # != override '
7680450520360_pi_fr: 'Dr. Reckeweg® R23 Nosoderm, gouttes buvables # != override '
7680687100014_pi_de: 'Comirnaty® 3 Mikrogramm/Dosis Konzentrat zur Herstellung einer Injektionsdispersion für Kinder ab 6 Monaten bis unter 5 Jahren (KASTANIENBRAUNE Kappe) # != override '
7680687100021_pi_de: 'Comirnaty® 3 Mikrogramm/Dosis Konzentrat zur Herstellung einer Injektionsdispersion für Kinder ab 6 Monaten bis unter 5 Jahren (KASTANIENBRAUNE Kappe) # != override '
7680687100052_pi_de: 'Comirnaty® 3 Mikrogramm/Dosis Konzentrat zur Herstellung einer Injektionsdispersion für Kinder ab 6 Monaten bis unter 5 Jahren (KASTANIENBRAUNE Kappe) # != override '
7680687100038_pi_de: 'Comirnaty® 3 Mikrogramm/Dosis Konzentrat zur Herstellung einer Injektionsdispersion für Kinder ab 6 Monaten bis unter 5 Jahren (KASTANIENBRAUNE Kappe) # != override '
7680687100045_pi_de: 'Comirnaty® 3 Mikrogramm/Dosis Konzentrat zur Herstellung einer Injektionsdispersion für Kinder ab 6 Monaten bis unter 5 Jahren (KASTANIENBRAUNE Kappe) # != override '
7680687100069_pi_de: 'Comirnaty® 3 Mikrogramm/Dosis Konzentrat zur Herstellung einer Injektionsdispersion für Kinder ab 6 Monaten bis unter 5 Jahren (KASTANIENBRAUNE Kappe) # != override '
7680687100014_pi_fr: 'Comirnaty® 3 microgrammes/dose dispersion à diluer pour dispersion injectable pour les enfants de 6 mois à moins de 5 ans (couvercle MARRON) # != override '
7680687100021_pi_fr: 'Comirnaty® 3 microgrammes/dose dispersion à diluer pour dispersion injectable pour les enfants de 6 mois à moins de 5 ans (couvercle MARRON) # != override '
7680687100052_pi_fr: 'Comirnaty® 3 microgrammes/dose dispersion à diluer pour dispersion injectable pour les enfants de 6 mois à moins de 5 ans (couvercle MARRON) # != override '
7680687100038_pi_fr: 'Comirnaty® 3 microgrammes/dose dispersion à diluer pour dispersion injectable pour les enfants de 6 mois à moins de 5 ans (couvercle MARRON) # != override '
7680687100045_pi_fr: 'Comirnaty® 3 microgrammes/dose dispersion à diluer pour dispersion injectable pour les enfants de 6 mois à moins de 5 ans (couvercle MARRON) # != override '
7680687100069_pi_fr: 'Comirnaty® 3 microgrammes/dose dispersion à diluer pour dispersion injectable pour les enfants de 6 mois à moins de 5 ans (couvercle MARRON) # != override '
```
